sdk/include/fail-simulator-on-error.h

```diff
@@ -93,5 +68,6 @@ compartment_error_handler(ErrorState *frame, size_t mcause, size_t mtval)
 		DebugErrorHandler::log("Unhandled error {} at {}", mcause, frame->pcc);
 		simulation_exit(1);
 	}
+
 	return ErrorRecoveryBehaviour::ForceUnwind;
```

Comment:
I don't think it matters but this return is now unreachable. You could bring `simulation_exit` below the `if`. It should be marked `noreturn`.

Reply:
I went back and forth on that and decided to leave it in case we stuck some other case arm in here again.  But that's pretty weak justification, so if you'd rather it be below, below it goes. :)

Reply:
If you’re running on something that *isn’t* a simulator, it is a local call that does nothing (and gets optimised away) so it should not be assumed to be noreturn.